Feature/homework 3

RinatShelkov opened the pull request.

RinatShelkov committed:
Add functions get_dict_with_key_state

RinatShelkov committed:
Add functions get_sorted_list

DwayneWard commented on src/processing.py:
```diff
@@ -0,0 +1,10 @@
+def get_dict_with_key_state(list_dict: list) -> list:
+    """Функция принимает список словарей и возвращает список словарей с ключом state = EXECUTED
+    :param list_dict
```
Описание параметров и возвращаемых значений нужно заполнить описанием. Например, 
`:param list_dict: Список словарей с операцийми для фильтрации`

DwayneWard commented on src/processing.py:
```diff
@@ -0,0 +1,10 @@
+    :param list_dict
+    :return list_executed"""
+
+    list_executed = []
```
Можно попробовать схлопнуть эту историю в list comrehantions

DwayneWard commented on src/processing.py:
```diff
@@ -8,3 +8,16 @@ def get_dict_with_key_state(list_dict: list) -> list:
+    отсортированы по убыванию даты(ключ date). Функция принмиает два аргумента, второй необязательный
+    задает порядок сортировки(убывание, возраcтание)
+    :param
+    list_dict
```
Тоже самое что в прошлом комменте :)

DwayneWard commented on src/processing.py:
```diff
@@ -0,0 +1,10 @@
+def get_dict_with_key_state(list_dict: list) -> list:
```
Нехватает атрибута со значением по умолчанию для указания ключа для фильтрации

DwayneWard reviewed (approved):
В целом, хорошо :)

DwayneWard commented on src/processing.py:
```diff
@@ -0,0 +1,10 @@
+
+    list_executed = []
+    for dictionary in list_dict:
+        if dictionary["state"] == "EXECUTED":
```
Функция должна уметь сортировать любые статусы. Поэтому нам и нужен аргумент в функции

RinatShelkov committed:
Added in docstring description of parameters and returned values functions get_dict_with_key_state
Add argument 'key' to function get_dict_with_key_state
Renaming the list returned by a function(list_comrehantions)
Added in docstring description of parameters and returned values functions get_sorted_list

DwayneWard commented on src/processing.py:
```diff
@@ -1,23 +1,27 @@
-            list_executed.append(dictionary)
-    return list_executed
+        if dictionary["state"] == key:
+            list_comrehantions.append(dictionary)
```
Когда я писал про LC я имел ввиду конструкцию генератора списков, а не переименование переменных :)

DwayneWard commented on src/processing.py:
```diff
@@ -1,23 +1,27 @@
-    list_dict
-    revers
-    :return sorted_list"""
+    list_dict: Список словарей с операциями для фильтрации по дате
```
Молодец!
